Some s390x related tweaks and one bugfix

ldoktor opened the pull request:
This addresses some issues related to s390x. I used RHEL.7.devel defined in https://github.com/avocado-framework/avocado-vt/pull/1200

luckyh commented:
@Zhengtong call for review, thanks.

Zhengtong commented on qemu/tests/cfg/multi_disk.cfg:
```diff
@@ -93,7 +93,10 @@
+                    !s390x:
+                        stg_params += "drive_unit:range(0,255,1,3) "
+                    s390x:
+                        stg_params += "drive_unit:range(0,8,1,3) "
```
Is there any technics background to set the max scsi-id of s390x to be less than 8 ?  The max scsi-id on s390x is still 255 now.

ldoktor replied:
This limit came from `virttest.qemu_devices.qcontainer` where in case of `ccw` the ranges were set according to `lsi` device. Anyway I just verified it works well with 255 range so moving this fix from here to https://github.com/avocado-framework/avocado-vt/pull/1200

ldoktor commented:
Removed the commit which changed multi_disk scsi-id ranges and fixed the virttest representation in https://github.com/avocado-framework/avocado-vt/pull/1200 instead. Now it should be all set.

ldoktor commented:
```
avocado run multi_disk..virtio_scsi_variants --vt-guest-os RHEL.7.devel.s390x --vt-extra-param
JOB ID     : 591b6ef909bf25a4fbbc601730a90790abd1c755
JOB LOG    : /root/avocado/job-results/job-2017-11-08T10.23-591b6ef/job.log
 (1/5) type_specific.io-github-autotest-qemu.multi_disk.virtio_scsi_variants.passthrough.block: PASS (112.53 s)
 (2/5) type_specific.io-github-autotest-qemu.multi_disk.virtio_scsi_variants.passthrough.generic: PASS (462.14 s)
 (3/5) type_specific.io-github-autotest-qemu.multi_disk.virtio_scsi_variants.multi_lun: ERROR (447.60 s)
 (4/5) type_specific.io-github-autotest-qemu.multi_disk.virtio_scsi_variants.multi_scsiid_lun: PASS (1256.55 s)
 (5/5) type_specific.io-github-autotest-qemu.multi_disk.virtio_scsi_variants.multi_bus_scsiid_lun: PASS (296.80 s)
RESULTS    : PASS 4 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 2590.59 s
JOB HTML   : /root/avocado/job-results/job-2017-11-08T10.23-591b6ef/results.html
```

(where the `multi_lun` crashes as it can't handle so many devices, will follow up with devs to address the issue)

ldoktor commented:
OK, I added several little fixes based on feedback from s390x devels. Please see the changes in individual commits. (not all tests are filtered, only those that "most probably" will never be supported)

Zhengtong commented:
LGTM  Acked-by: Liu Zhengtong <zhengtli@redhat.com>
